astrobee/survey_manager/scripts/command_astrobee

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2021, United States Government, as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# 
+# All rights reserved.
+# 
+# The "ISAAC - Integrated System for Autonomous and Adaptive Caretaking
+# platform" software is licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import subprocess
+import sys
+
+
+
+# command = 'rosrun inspection inspection_tool -panorama -panorama_poses /resources/panorama_iss.txt -panorama_mode "5_mapper_and_hugin"'
+
+
+def send_command_with_input(command):
+    print(command)
+
+    # Start the process
+    process = subprocess.Popen(command, shell=True, stdin=subprocess.PIPE, stdout=sys.stdout, stderr=subprocess.PIPE, text=True)
+
+    try:
+        while process.poll() is None:
+
+            # Get user input dynamically
+            user_input = input()
+
+            # Check if the user wants to exit
+            if user_input.lower().strip() == 'exit':
+                break
+
+            # Send the user input to the process
+            process.stdin.write(user_input + "\n")
+            process.stdin.flush()
+
+        # Close stdin to indicate no more input will be sent
+        process.stdin.close()
+
+        # Read output and error (if any)
+        output, error = process.communicate()
+
+        # Get the final exit code
+        return  process.returncode
+
+    except:
+        return -1
+
+
+def send_command(command):
+    print(command)
+    process = subprocess.Popen(command, shell=True, stdout=sys.stdout, stderr=sys.stdout, text=True)
```

Comment:
subprocess.run() doesn't support the stdin=subprocess.PIPE

Reply:
I didn't mean to say you should change to `subprocess.run()` everywhere. But I think you can in this case because you are not using `stdin=subprocess.PIPE`. It's a minor simplification.